### PR TITLE
Task/video 10635/android integrate api call and customize UI for speakers

### PR DESCRIPTION
**Pull request by alejo740**
### JIRA link(s):
- [VIDEO-10635](https://issues.corp.twilio.com/browse/VIDEO-10635)

### Description
- Implemented API call to Join Stream As Speaker.
- Connected to a Room as speaker.
- Connected UI to turn on/off the camera and mic.
- When the room is disconnected, then show an Alert.

### Dependency:
This PR is depending on https://github.com/twilio/twilio-live-interactive-video/pull/153 to be merged. 

### Demo

https://user-images.githubusercontent.com/929504/182400586-88ef1286-dc22-4a3e-a2dd-4d31d28eac83.mp4

### CODE REVIEW

**Review by afalls-twilio — APPROVED** [review]

File: apps/android/LiveVideoApp/app/src/main/java/com/twilio/livevideo/app/manager/room/RoomManager.kt [hunk 1]
```diff
@@ -108,23 +113,28 @@ class RoomManager @Inject constructor(
...
-
+        Timber.i("onReconnected -> room sid: %s", room.sid)
     }
 
     override fun onDisconnected(room: Room, twilioException: TwilioException?) {
-
+        Timber.i("onDisconnected -> room sid: %s", room.sid)
+        if (!isDisconnectedFromUI) {
+            disconnect(false)
+        }
```
Comment (timrozum) [inline apps/android/LiveVideoApp/app/src/main/java/com/twilio/livevideo/app/manager/room/RoomManager.kt:133]: I think we need to look for specific errors here to explicitly detect when a room is ended by the host. 

See this example: https://github.com/twilio/twilio-live-interactive-video/blob/6d01822ff59c6666754375472043b2270e40237b/apps/ios/LiveVideo/LiveVideo/Twilio/Room/RoomManager.swift#L86

File: apps/android/LiveVideoApp/app/src/main/java/com/twilio/livevideo/app/manager/room/RoomManager.kt [hunk 2]
```diff
@@ -65,16 +66,20 @@ class RoomManager @Inject constructor(
         }
     }
 
-    fun disconnect() {
-        handleError(null)
+    fun disconnect(fromUI: Boolean = true) {
```
Comment (timrozum) [inline apps/android/LiveVideoApp/app/src/main/java/com/twilio/livevideo/app/manager/room/RoomManager.kt:69]: I think this function should only be called from the UI and the parameter can then be removed.

**Review by timrozum — CHANGES_REQUESTED** [review]

File: apps/android/LiveVideoApp/app/src/main/java/com/twilio/livevideo/app/manager/room/RoomManager.kt (hunk 2 above)
Comment (alejo740) [reply apps/android/LiveVideoApp/app/src/main/java/com/twilio/livevideo/app/manager/room/RoomManager.kt]: Done!

File: apps/android/LiveVideoApp/app/src/main/java/com/twilio/livevideo/app/manager/room/RoomManager.kt (hunk 1 above)
Comment (alejo740) [reply apps/android/LiveVideoApp/app/src/main/java/com/twilio/livevideo/app/manager/room/RoomManager.kt]: Thanks! This help me a lot to clarify how to handle multiple disconnection types.

File: apps/android/LiveVideoApp/app/src/main/java/com/twilio/livevideo/app/manager/room/RoomManager.kt [hunk 3]
```diff
@@ -65,16 +65,19 @@ class RoomManager @Inject constructor(
         }
     }
 
-    fun disconnect() {
-        handleError(null)
+    fun disconnect(disconnectionType: RoomDisconnectionType? = null) {
```
[the remark this answers is not in the document]
Comment (alejo740) [reply apps/android/LiveVideoApp/app/src/main/java/com/twilio/livevideo/app/manager/room/RoomManager.kt]: Done!

**Review by timrozum — APPROVED** [review]

**Pull request merged**